src/libraries/System.Private.CoreLib/src/System/Threading/TimeoutHelper.cs

```diff
@@ -49,5 +49,26 @@ public static int UpdateTimeOut(uint startTime, int originalWaitMillisecondsTime
 
             return currentWaitTimeout;
         }
+
+        /// <summary>
+        /// Helper function to measure and update the elapsed time
+        /// </summary>
+        /// <param name="startTime"> The first time (in milliseconds) observed when the wait started</param>
+        /// <param name="originalWaitMillisecondsTimeout">The original wait timeout in milliseconds</param>
+        /// <returns>The new wait time in milliseconds</returns>
+        public static uint UpdateTimeOut(uint startTime, uint originalWaitMillisecondsTimeout)
+        {
+            // The function must be called in case the time out is not infinite
+            Debug.Assert(originalWaitMillisecondsTimeout != Timeout.UnsignedInfinite);
+
+            uint elapsedMilliseconds = GetTime() - startTime;
```

Comment:
Should this use long and `Environment.TickCount64` to avoid integer overflow issues when the wait time is close to the max value?

Reply:
Yes I think that'd be better